src/default_client.rs

```diff
@@ -12,6 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! The default implementation of the Nakama [`Client`] trait.
+//!
+//! # General concepts
```

Comment:
@seanpackham @gpene what are you guys' thoughts on inlining these sort of concepts? Should @marot instead point out to a documentation link?

Reply:
@lugehorsam I think these klind of usage examples go in the documentation like they have for our other client libraries. It's up to you guys where it should be duplicated (lightly or completely) or linked to from the code.